### PR TITLE
bpf: Add macro for socket_filter programs

### PR DESCRIPTION
**Pull request by dave-tucker**
Signed-off-by: Dave Tucker <dave@dtucker.co.uk>

### CODE REVIEW

File: bpf/aya-bpf-macros/src/lib.rs [hunk 1]
```diff
@@ -248,3 +248,31 @@ pub fn btf_tracepoint(attrs: TokenStream, item: TokenStream) -> TokenStream {
         .unwrap_or_else(|err| err.to_compile_error())
         .into()
 }
+
+/// Marks a function as a eBPF Socket Filter program that can be attached at
+/// a raw AF_PACKET socket.
```
Comment (alessandrod) [inline bpf/aya-bpf-macros/src/lib.rs:253]: Socket filters can be attached to other sockets too not just AF_PACKET

**Review by alessandrod — APPROVED** [review]

File: bpf/aya-bpf-macros/src/lib.rs (hunk 1 above)
Comment (dave-tucker) [reply bpf/aya-bpf-macros/src/lib.rs]: fixed it to say "attached to a raw socket"

File: bpf/aya-bpf-macros/src/lib.rs [hunk 2]
```diff
@@ -248,3 +248,31 @@ pub fn btf_tracepoint(attrs: TokenStream, item: TokenStream) -> TokenStream {
         .unwrap_or_else(|err| err.to_compile_error())
         .into()
 }
+
+/// Marks a function as a eBPF Socket Filter program that can be attached to
+/// a raw socket.
```
Comment (alessandrod) [inline bpf/aya-bpf-macros/src/lib.rs:253]:
```suggestion
/// a socket.
```

Comment (dave-tucker) [reply bpf/aya-bpf-macros/src/lib.rs]: Fixed!

**Pull request merged**